src/typechecker/FStar.TypeChecker.TcTerm.fs

```diff
@@ -1203,6 +1204,8 @@ and tc_constant (env:env_t) r (c:sconst) : typ =
                                  (Const.const_to_string c)) r
 
       | _ -> raise_error (Errors.Fatal_UnsupportedConstant, "Unsupported constant") r
+  in
+  { res with pos = r }
```

Comment:
It's a little surprising to tag the type with the position information from the constant. We don't do this elsewhere when computing types. Maybe the thing to do here should be in analogy with `Env.try_lookup_lid`

```
let try_lookup_lid env l =
    match try_lookup_lid_aux None env l with
    | None -> None
    | Some ((us, t), r) ->
      let use_range = range_of_lid l in
      let r = Range.set_use_range r (Range.use_range use_range) in
      Some ((us, Subst.set_use_range use_range t), r)
```
Where we set only the use range of the returned type.

Also see step 3 here https://github.com/FStarLang/FStar/wiki/Tracking-and-reporting-source-locations-in-SMT-errors